### PR TITLE
Optimize the execution conditions of the integration test script

### LINKED ISSUE #1578: Optimize the action script of the integration test
### What would you like to be enhanced?

1、Modify the integration test script and adjust the script execution conditions.

### Why is this needed?

1、No function modification is involved, and the script is still executed. As a result, the overall execution time of the integration test script is too long.

### PR DESCRIPTION
**Pull request by hanbingleixue**
**What type of PR is this?**

Integration Test.

**What this PR does / why we need it?**

1、Optimize the execution conditions of the integration test script，the integration test script of the object is executed only when the related function is modified.

**Which issue(s) this PR fixes？**

Fixes #1578

**Does this PR introduce a user-facing change?**

No

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/sermant-io/Sermant/issues) related with this PR before you start working on it.
- [x] Make sure you have squashed your change to one single commit.
- [x] GitHub Actions works fine in this PR.

### CODE REVIEW

File: .github/actions/scenarios/spring/graceful/action.yml [hunk 1]
```diff
@@ -59,7 +59,7 @@ runs:
       run: |
         ps -ef | grep java
         bash ./sermant-integration-tests/scripts/checkService.sh http://127.0.0.1:8015/graceful/testGraceful 120
-        bash ./sermant-integration-tests/scripts/checkService.sh http://127.0.0.1:8005/graceful/testGraceful 60
+        bash ./sermant-integration-tests/scripts/checkService.sh http://127.0.0.1:8005/graceful/testGraceful 120
```
Comment (lilai23) [inline .github/actions/scenarios/spring/graceful/action.yml]: why is this changed to 120

File: .github/actions/common/plugin-change-check/action.yml [hunk 2]
```diff
@@ -76,26 +76,456 @@ runs:
...
+        echo "enableDynamicConfig=${{ steps.changed-dynamic-config.outputs.changed }}" >> $GITHUB_ENV
     - uses: marceloprado/has-changed-path@v1.0.1
       id: changed-workflow-or-test
       with:
-        paths: ./.github/actions ./.github/workflows sermant-integration-tests
+        paths: ./.github/actions/common ./.github/workflows sermant-integration-tests
+    - uses: marceloprado/has-changed-path@v1.0.1
+      id: changed-workflow
+      with:
+        paths: ./.github/actions ./.github/workflows
```
Comment (lilai23) [inline .github/actions/common/plugin-change-check/action.yml]: not correct

File: .github/workflows/tagtransmission_integration_test.yml [hunk 3]
```diff
@@ -120,6 +123,7 @@ jobs:
       - name: common operations
         uses: ./.github/actions/common/tag-transmission
       - name: tag transmission test for httpClientVersion=${{ matrix.httpClientVersion }}
+        if: env.enableTagTransmission_HttpclientV3=='true'
```
Comment (lilai23) [inline .github/workflows/tagtransmission_integration_test.yml]: add set-testcase-execution job

File: .github/actions/common/plugin-change-check/action.yml [hunk 4]
```diff
@@ -46,15 +46,15 @@ runs:
     - uses: marceloprado/has-changed-path@v1.0.1
       id: changed-sermant-monitor
       with:
-        paths: sermant-plugins/sermant-monitor
+        paths: sermant-plugins/sermant-monitor ./.github/actions/scenarios/dubbo/monitor
```
Comment (lilai23) [inline .github/actions/common/plugin-change-check/action.yml:49]: add ./.github/actions/common/monitor

File: .github/actions/common/plugin-change-check/action.yml [hunk 5]
```diff
@@ -76,26 +76,456 @@ runs:
...
+        paths: sermant-plugins/sermant-service-removal ./.github/actions/scenarios/dubbo/removal
+    - name: env sermant-sermant-service-removal
       shell: bash
       run: |
-        echo "sermantMqConsumeProhibitionChanged=${{ steps.changed-sermant-mq-consume-prohibition.outputs.changed
-        }}" >> $GITHUB_ENV
+        echo "sermantServiceRemovalChanged=${{ steps.changed-sermant-removal.outputs.changed }}" >> $GITHUB_ENV
     - uses: marceloprado/has-changed-path@v1.0.1
-      id: changed-sermant-removal
+      id: changed-sermant-agentcore-no-dynamicConfig
```
Comment (lilai23) [inline .github/actions/common/plugin-change-check/action.yml]: add changed-sermant-agentcore-dynamicConfig

File: .github/actions/common/plugin-change-check/action.yml [hunk 6]
```diff
@@ -76,26 +76,456 @@ runs:
...
+      run: |
+        echo "enableTagTransmission_crossthread=${{ steps.changed-tag-transmission-crossthread.outputs.changed }}" >> $GITHUB_ENV
+    - uses: marceloprado/has-changed-path@v1.0.1
+      id: changed-dubbo-router
+      with:
+        paths: ./.github/actions/scenarios/dubbo/router
+    - name: env dubbo-router
+      shell: bash
+      run: |
+        echo "enableDubboRouter=${{ steps.changed-dubbo-router.outputs.changed }}" >> $GITHUB_ENV
```
Comment (lilai23) [inline .github/actions/common/plugin-change-check/action.yml]: changed-sermant-spring-router
changed-sermant-dubbo-router

File: .github/actions/scenarios/spring/graceful/action.yml (hunk 1 above)
Comment (hanbingleixue) [reply .github/actions/scenarios/spring/graceful/action.yml]: done

File: .github/actions/common/plugin-change-check/action.yml (hunk 2 above)
Comment (hanbingleixue) [reply .github/actions/common/plugin-change-check/action.yml]: done

File: .github/workflows/tagtransmission_integration_test.yml (hunk 3 above)
Comment (hanbingleixue) [reply .github/workflows/tagtransmission_integration_test.yml]: done

File: .github/actions/common/plugin-change-check/action.yml (hunk 4 above)
Comment (hanbingleixue) [reply .github/actions/common/plugin-change-check/action.yml:49]: add enableSpringCommonActionChange and enableDubboCommonActionChange to determine the common scripts of Spring and dubbo.

File: .github/actions/common/plugin-change-check/action.yml (hunk 5 above)
Comment (hanbingleixue) [reply .github/actions/common/plugin-change-check/action.yml]: No, just changed-sermant-agentcore-no-dynamicConfig

File: .github/actions/common/plugin-change-check/action.yml (hunk 6 above)
Comment (hanbingleixue) [reply .github/actions/common/plugin-change-check/action.yml]: done

File: .github/workflows/agentcore_integration_test.yml [hunk 7]
```diff
@@ -11,11 +11,32 @@ on:
...
   group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}-${{ github.head_ref }}
   cancel-in-progress: true
 jobs:
+  set-execution-conditions:
+    name: set-execution-conditions
+    runs-on: ubuntu-latest
+    outputs:
+      enableAgentCoreNoDynamicConfig: ${{ steps.set-outputs.outputs.enableAgentCoreNoDynamicConfig }}
+      enableAllAction: ${{ steps.set-outputs.outputs.enableAllAction }}
+    steps:
```
Comment (lilai23) [inline .github/workflows/agentcore_integration_test.yml]: put outputs behing steps

File: .github/workflows/spring_integration_test_1.yml [hunk 8]
```diff
@@ -139,23 +177,23 @@ jobs:
       - name: common operations
         uses: ./.github/actions/common/spring
       - name: (graceful) test for springboot=${{ matrix.springBootVersion }} springCloudVersion=${{ matrix.springCloudVersion }}
-        if: env.enableGraceful == 'true'
+        if: needs.set-execution-conditions.outputs.enableAllAction == 'true' || needs.set-execution-conditions.outputs.enableGraceful == 'true'
```
Comment (lilai23) [inline .github/workflows/spring_integration_test_1.yml]: put enableAllAction into pluginchangecheck

**Review by daizhenyu — DISMISSED** [review]

**Review by lilai23 — APPROVED** [review]

**Review by daizhenyu — APPROVED** [review]
